### PR TITLE
KUBESAW-43: Replace the special ToolchainClusterCondition with the standard toolchain Condition

### PR DESCRIPTION
**Pull request by fbm3307**
This is to replace the special ToolchainClusterCondition with the standard toolchain Condition

This is Related to Change in

- API - https://github.com/codeready-toolchain/api/pull/416
- Toolchain-Common - https://github.com/codeready-toolchain/toolchain-common/pull/391
- Member-Operator - https://github.com/codeready-toolchain/member-operator/pull/560
- Toolchain-E2E - https://github.com/codeready-toolchain/toolchain-e2e/pull/955
- KSCTL - https://github.com/kubesaw/ksctl/pull/38

Only change is updating the dependency of API and toolchain-common

- Registration-Service - https://github.com/codeready-toolchain/registration-service/pull/424
- Sandbox-SRE - https://github.com/codeready-toolchain/sandbox-sre/pull/1706

### CODE REVIEW

**Review by mfrancisc — APPROVED** [review]
Nice Job 👍

**Comment by rsoaresd** [comment]
Hello @fbm3307 👋

All the runs of this PR failed with `reached timeout of waiting for CRD memberoperatorconfigs.toolchain.dev.openshift.com to be available in the cluster and the InstallPlan to be complete`.

Very likely is PR specific. At least, from this [search query](https://search.dptools.openshift.org/?search=reached+timeout+of+waiting+for+CRD+memberoperatorconfigs.toolchain.dev.openshift.com+to+be+available+in+the+cluster+and+the+InstallPlan+to+be+complete&maxAge=336h&context=1&type=build-log&name=.*codeready-toolchain.*&excludeName=&maxMatches=5&maxBytes=20971520&groupBy=job), it seems.

**Comment by mfrancisc** [comment]
Maybe we need first to make that field optional and then migrate to the new `Condition` ? Not sure if that will workaround the issue.

**Comment by fbm3307** [comment]
Thanks @rsoaresd for lookinng into the errors. 
@mfrancisc yes i think, will make `LastProbetTime` Optional first , also add the new field of `LastTransitionTime` then switch to new condition

**Comment by fbm3307** [comment]
looks like a flaky test 
/test e2e

**Comment by fbm3307** [comment]
looks like the e2e failure is not related to this PR

File: controllers/toolchainstatus/toolchainstatus_controller_test.go [hunk 1]
```diff
@@ -1835,10 +1835,9 @@ func cachedToolchainCluster(cl runtimeclient.Client, name string, status corev1.
 		},
 		Client: cl,
 		ClusterStatus: &toolchainv1alpha1.ToolchainClusterStatus{
-			Conditions: []toolchainv1alpha1.ToolchainClusterCondition{{
+			Conditions: []toolchainv1alpha1.Condition{{
 				Type:            toolchainv1alpha1.ToolchainClusterReady,
```
Comment (MatousJobanek) [inline controllers/toolchainstatus/toolchainstatus_controller_test.go:1839]: As mentioned in the other PR, we could replace the special condition type with the generic one - there is no need to have a special type anymore

Comment (fbm3307) [reply controllers/toolchainstatus/toolchainstatus_controller_test.go]: done

**Review by metlos — APPROVED** [review]

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**